### ISSUE TITLE
[Improvement]: Add support to `gotenberg/gotenberg-php:2.0.0`

### ISSUE THREAD

**Issue by isekep** [pimcore/pimcore#16566]
### Pimcore version

11.1.5

### Steps to reproduce

1. Install Pimcore (composer require pimcore/pimcore:11.1.5 pimcore/admin-classic-ui-....... etc.)
2. Add some packages including latest Gotenberg PHP lib to installation (composer require gotenberg/gotenberg-php:*)
3. Open Admin backend and try to open an asset or to navigate through the asset tree

### Actual Behavior

Error popup is shown:
> Attempted to call an undefined method named "html" of class "Gotenberg\Modules\Chromium".

Navigation through asset tree is broken / asset itself cannot be opened

----

Reason:
Latest gotenberg/gotenberg-php:2.0.0 does not offer the `html()`method anymore, but Pimcore 11.1.5 is calling it in `pimcore/vendor/pimcore/pimcore/lib/Document/Adapter/Gotenberg.php` in line 72:
```php
$request = GotenbergAPI::chromium(Config::getSystemConfiguration('gotenberg')['base_url'])
            ->html(Stream::string('dummy.html', '<body></body>'));
```
Gotenberg PHP lib v2.0.0 was released early in January 2024 and its release note hints some BREAKING CHANGES. 

Workaround:
Downgrade to latest 1.x version of Gotenberg PHP lib: `composer require gotenberg/gotenberg-php:^1.1`.

### Expected Behavior

Pimcore is fully supporting Gotenberg PHP lib v2.0.0 (preferred) -- or has at least a constraint which prevents Gotenberg PHP lib v2+ to be installed.

**Comment by kingjia90** [comment]
https://github.com/gotenberg/gotenberg-php/releases/tag/v2.0.0 got released on Jan 4 and we were not prepared for that 😄 
SImilarly to https://github.com/pimcore/web-to-print-bundle/pull/57 

So far, Pimcore only supports `^1.1`, will "fix" the composer suggest and docs (#16569) , but ~looking forward for a PR~(~i think i'll do it~ nope, Bernie did that 😄 ) need to extend the support to `2.0` for next minor release

**Comment by kingjia90** [comment]
Extra notes: The [breaking change](https://github.com/gotenberg/gotenberg-php/releases#:~:text=The%20PDF%20/%20Screenshot%20features%20are%20now%20available%20via%20Gotenberg%3A%3Achromium(%24apiUrl)%2D%3Epdf()%2D%3E...%20and%20Gotenberg%3A%3Achromium(%24apiUrl)%2D%3Escreenshot()%2D%3E...) of v2 is that instead of directly `->html()`, we have a method to decide wheter generate a pdf or screenshot `pdf`/`screenshot` before that, eg.`pdf()->html()` or `screenshot()->html()` 

The good thing is now we can generate preview images without having a pdf in-between converted to image 
https://github.com/pimcore/pimcore/pull/14212#discussion_r1115942700 and would need a follow-up

Update: noticed https://github.com/pimcore/pimcore/issues/16605 got open 👍🏻 linking the 2 improvement issue as related
